### PR TITLE
Added a C example according to the issue 'C Example #357'

### PR DESCRIPTION
**Pull request by Gauravsaha-97**

### CODE REVIEW

**Comment by alexcb** [comment]
Thanks for the contribution, ideally we're looking for an example that would have two or more files that get linked together along with a library. We would then need a corresponding Earthfile to be written which would compile the program. Take a look at our existing c++ example: https://github.com/earthly/earthly/tree/master/examples/cpp

What would be really cool is if we could have two binaries build: one which is the main example, and a second binary which does some unit testing.

**Comment by Gauravsaha-97** [comment]
I have tried to implement what you have mentioned before. I am not very confident as I am doing this type of task for the first time. I would be happy to rectify in case of any issue.

File: examples/c/src/main.c [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#include <stdio.h>
+
+#include "sum.h"
+
+int main(int argc, char *argv[]) {
+	int n1 = 100;
+    int n2 = 200;
```
Comment (alexcb) [inline examples/c/src/main.c:7]: It looks like there's a mix between tabs and spaces in this file; can you change this to consistently use one or the other.

File: examples/c/src/sum.c [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#include "sum.h"
+
+int sum(int n1, int n2)
+{
```
Comment (alexcb) [inline examples/c/src/sum.c:4]: The style of this function is different from the style used in the main function (i.e. the main function has the curly brace on the same line as the function, but this one is on the second line). Can you pick a single style and use it consistently.

**Review by alexcb — CHANGES_REQUESTED** [review]
Thanks for the changes, they are looking good, left a few nitpicky issues related to the consistently using the same code style.

Would you be able to also write a test for the `sum` function using the cunit library, which can be installed via `apt-get install libcunit1-dev`.

File: examples/c/src/main.c (hunk 1 above)
Comment (Gauravsaha-97) [reply examples/c/src/main.c]: Okay..I will fix them.

**Comment by Gauravsaha-97** [comment]
@alexcb Added a new file which contains the unit tests.

**Comment by alexcb** [comment]
This is looking great! Thanks for adding a sample unit test. How does one run the unit test? Can we add those steps to the Earthfile under a `test` target?

**Comment by Gauravsaha-97** [comment]
@alexcb Added the testing procedures for all modes.

**Comment by alexcb** [comment]
I'm not sure those additions to the Earthfile are going to work; did you test them by running `earth +test`?

**Comment by Gauravsaha-97** [comment]
@alexcb I have added the necessary Header Files to the directory. If something is not right, can you please suggest some resources to learn about it?

**Comment by alexcb** [comment]
https://docs.earthly.dev/guides/basics is a good reference to learn about earthly. It shows some basic hello world examples for different languages.

I don't think adding the cunit source code is needed; instead it should be possible to install the `libcunit1-dev` package via a `RUN` command in the Earthfile.

**Pull request closed without merging**